lisa/sut_orchestrator/azure/features.py

```diff
@@ -2517,9 +2519,10 @@ def on_before_deployment(cls, *args: Any, **kwargs: Any) -> None:
             if "platformUpdateDomainCount" not in params.availability_set_properties:
                 params.availability_set_properties["platformUpdateDomainCount"] = 1
         elif params.availability_type == AvailabilityType.AvailabilityZone:
-            assert (
-                params.availability_zones
-            ), "Availability Zone is selected, but no zone was provided."
+            if not params.availability_zones:
+                raise SkippedException(
+                    "Availability Zone is selected, but no zone was provided."
```

Comment:
Please refine the messages clearer. Where the zone is selected, and where it's not provided?

Reply:
Fixed

Reply:
How about the source of first half sentence? where the zone is selected from?

Reply:
Technically it could have been from a test case requirement but very likely it was also from the runbook.

With these error messages, it is hard to think of every single combination of (runbook, test case, vm capabilities) × (disk type, maximize capability, availability type, tip session, zones)

I wrote this logic to verify that all the setting look correct and have appropriate defaults, agnostic of where the requirements are coming from. Because most of the requirement merging happens before this logic in `on_before_deployment`

Reply:
I have changed the error message to include explicit troubleshooting steps.